test/Microsoft.Health.Dicom.Tests.Integration/Features/CustomTagServiceTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.CustomTag;
+using Microsoft.Health.Dicom.Core.Messages.CustomTag;
+using Microsoft.Health.Dicom.Tests.Common.Extensions;
+using Microsoft.Health.Dicom.Tests.Integration.Persistence;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Tests.Integration.Features
+{
+    public class CustomTagServiceTests : IClassFixture<CustomTagServiceTestsFixture>
+    {
+        private static readonly CancellationToken DefaultCancellationToken = new CancellationTokenSource().Token;
+
+        private readonly CustomTagServiceTestsFixture _customTagServiceTestsFixture;
+
+        public CustomTagServiceTests(CustomTagServiceTestsFixture customTagServiceTestsFixture)
+        {
+            EnsureArg.IsNotNull(customTagServiceTestsFixture, nameof(customTagServiceTestsFixture));
+            _customTagServiceTestsFixture = customTagServiceTestsFixture;
+        }
+
+        [Fact(Skip = "Feature not ready")]
+        public async Task GivenValidCustomTags_WhenAddCustomTag_ThenTagShouldBeAdded()
+        {
+            DicomTag tag1 = DicomTag.DeviceSerialNumber;
```

Comment:
A couple questions:

1. Do we need to handle multiple instances of our integration tests running? Would that cause problems?
2. We are using `finally` blocks to clean up our tests should they fail, but do we need to worry about if the process stops/fails in them middle of a test such that the DBs are not clean?

Reply:
1. are you talking about each fixture having one database instance? 
2. when test fail, the db stays for diagnosis. the db will be clean up when the PR is closed by Github Action https://github.com/microsoft/dicom-server/actions?query=workflow%3ACleanupAzureResources